core/tracker/tracker.go

```diff
@@ -42,11 +46,15 @@ const (
 	sentinel
 )
 
+// shareIdx represents a peer's share index.
+type shareIdx int
```

Comment:
rather not introduce this type, naming `var shareIdx int` is what we use through the rest of the codebase.

Reply:
yeah, i introduced this type to avoid any confusion with int and shareIdx since `shareIdx` is used in this file very extensively